activiti-cloud-services-runtime-bundle/activiti-cloud-services-rest-impl/src/test/java/org/activiti/cloud/services/rest/controllers/ProcessVariablesHelperIT.java

```diff
@@ -0,0 +1,208 @@
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.activiti.cloud.services.rest.controllers;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.BDDMockito.given;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.activiti.api.process.model.builders.ProcessPayloadBuilder;
+import org.activiti.api.runtime.conf.impl.CommonModelAutoConfiguration;
+import org.activiti.api.runtime.conf.impl.ProcessModelAutoConfiguration;
+import org.activiti.cloud.services.events.ProcessEngineChannels;
+import org.activiti.cloud.services.events.configuration.CloudEventsAutoConfiguration;
+import org.activiti.cloud.services.events.configuration.RuntimeBundleProperties;
+import org.activiti.cloud.services.rest.conf.ServicesRestAutoConfiguration;
+import org.activiti.engine.ActivitiException;
+import org.activiti.spring.process.conf.ProcessExtensionsAutoConfiguration;
+import org.activiti.spring.process.model.Extension;
+import org.activiti.spring.process.model.ProcessExtensionModel;
+import org.activiti.spring.process.model.VariableDefinition;
+import org.activiti.spring.process.variable.VariableValidationService;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.restdocs.AutoConfigureRestDocs;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Import;
+import org.springframework.data.web.config.EnableSpringDataWebSupport;
+import org.springframework.test.context.junit4.SpringRunner;
+
+@RunWith(SpringRunner.class)
+@WebMvcTest(ProcessVariablesHelper.class)
+@EnableSpringDataWebSupport
+@AutoConfigureMockMvc(secure = false)
+@AutoConfigureRestDocs(outputDir = "target/snippets")
+@Import({CommonModelAutoConfiguration.class,
+        ProcessModelAutoConfiguration.class,
+        RuntimeBundleProperties.class,
+        CloudEventsAutoConfiguration.class,
+        ProcessExtensionsAutoConfiguration.class,
+        ServicesRestAutoConfiguration.class})
+@ComponentScan(basePackages = {"org.activiti.cloud.services.rest.assemblers", "org.activiti.cloud.alfresco"})
+public class ProcessVariablesHelperIT {
```

Comment:
IMO, we don't need to cover this class with integration tests. We can cover it using unit tests (mocking `VariableValidationService` and `DateFormatterProvider`) and have only one integration tests per impacted endpoint in the starter covering cases of error.

Reply:
It is already there :-)... Should I remove it?

Reply:
Let's keep it.